mono/utils/mono-mmap.h

```diff
@@ -65,6 +65,12 @@ MONO_API int   mono_file_unmap_fileio (void *addr, void *handle);
 #endif
 MONO_API int   mono_mprotect   (void *addr, size_t length, int flags);
 
+// Last two parameters are optional.
+// This is mono_file_map but with optionally returning an error message.
+// See https://github.com/mono/mono/issues/8225.
+MONO_API void*
+mono_file_map_error (size_t length, int flags, int fd, guint64 offset, void **ret_handle, const char *filepath, char **error_message);
```

Comment:
Not sure we should introduce a new public API for this. Couldn't we just do the error formatting on the caller side?

Reply:
If we assume the failure ended by setting win32 lasterror. Not clearly a good binding to have.